client/src/main/java/com/netflix/conductor/client/automator/TaskPollExecutor.java

```diff
@@ -162,38 +170,47 @@ void pollAndExecute(Worker worker) {
                                                     worker.getBatchPollTimeoutInMS()));
             acquiredTasks = tasks.size();
             for (Task task : tasks) {
-                if (Objects.nonNull(task) && StringUtils.isNotBlank(task.getTaskId())) {
-                    MetricsContainer.incrementTaskPollCount(taskType, 1);
-                    LOGGER.debug(
-                            "Polled task: {} of type: {} in domain: '{}', from worker: {}",
-                            task.getTaskId(),
-                            taskType,
-                            domain,
-                            worker.getIdentity());
-
-                    CompletableFuture<Task> taskCompletableFuture =
-                            CompletableFuture.supplyAsync(
-                                    () -> processTask(task, worker, pollingSemaphore),
-                                    executorService);
-
-                    if (task.getResponseTimeoutSeconds() > 0 && worker.leaseExtendEnabled()) {
-                        ScheduledFuture<?> leaseExtendFuture =
-                                leaseExtendExecutorService.scheduleWithFixedDelay(
-                                        extendLease(task, taskCompletableFuture),
-                                        Math.round(
-                                                task.getResponseTimeoutSeconds()
-                                                        * LEASE_EXTEND_DURATION_FACTOR),
-                                        Math.round(
-                                                task.getResponseTimeoutSeconds()
-                                                        * LEASE_EXTEND_DURATION_FACTOR),
-                                        TimeUnit.SECONDS);
-                        leaseExtendMap.put(task.getTaskId(), leaseExtendFuture);
+                Map<String, String> headers = new HashMap<>();
```

Comment:
- Extract this to a separate method
- We can add an around aspect for adding the trace. KIndly check that